### PR TITLE
[FIX] payment: make _disableButton act only within the current widget

### PR DESCRIPTION
**Pull request by maneandrea**
The function _disableButton is defined as an arrow function (args) => {body;}. This assigns the `this` keyword to the global document instead of the class instance.

As a result the jQuery results for
```
const $submitButton = this.$('button[name="o_payment_submit_button"]');
```
are inconsistent between _enableButton and _disableButton. This could result in buttons being disabled and, in principle, never re-enabled.

**Current behavior before PR**:
Having the same widget call _disableButton and _enableButton might disable a button outside the widget and never re-enable it

**Desired behavior after PR is merged**:
If a button is disabled by a widget calling _disableButton it will get re-enabled too when _enableButton is applied.

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by JVegaB** [comment]
This LGTM +1, we need the function keyword to declare the new `this` context.

**Review by luisg123v — COMMENTED** [review]
I think title says the how but it should say the what. In other words, commit title should summarice what's being fixed, and commit description could extend that summary and also explaing how and why.

**Comment by fernandahf** [comment]
cc @AntoineVDV, @Feyensv

**Comment by moylop260** [comment]
What do you think @tivisse @bouvyd ?

**Review by AntoineVDV — APPROVED** [review]
Hi @maneandrea, thanks for the fix :)

Before we can merge, please rebase your branch onto 16.0 to include some fix commits that have recently been merged and will make your PR pass the CI checks.

@robodoo ignore
:point_up_2: The payment form JS has been completely rewritten in 17.0, so we don't need to forward-port this fix.

About request for reviews: you no longer need to manually tag reviewers because it is done automatically with our in-house codeowners system. Every file of the codebase is assigned to a given team:

image

**Comment by AntoineVDV** [comment]
@robodoo r+

**Pull request closed without merging**
